lib/jekyll/external.rb

```diff
@@ -52,24 +52,34 @@ def version_constraint(gem_name)
       #
       # names - a string gem name or array of gem names
       #
+      # rubocop:disable Metrics/MethodLength
       def require_with_graceful_fail(names)
         Array(names).each do |name|
           begin
             Jekyll.logger.debug "Requiring:", name.to_s
             require name
-          rescue LoadError => e
-            Jekyll.logger.error "Dependency Error:", <<~MSG
-              Yikes! It looks like you don't have #{name} or one of its dependencies installed.
-              In order to use Jekyll as currently configured, you'll need to install this gem.
-
-              The full error message from Ruby is: '#{e.message}'
-
-              If you run into trouble, you can find helpful resources at https://jekyllrb.com/help/!
-            MSG
-            raise Jekyll::Errors::MissingDependencyException, name
+          rescue LoadError
+            if name.include?("-")
+              require name.tr("-", "/")
+            else
+              raise
+            end
           end
         end
+      rescue LoadError => e
+        e.message =~ %r! -- (.+)$!i
+        specimen = Regexp.last_match[1]
+        Jekyll.logger.error "Dependency Error:", <<-MSG
+          Yikes! It looks like you don't have #{specimen} or one of its dependencies installed.
```

Comment:
Let's use `name` instead of `specimen` in this line.

`name` should still have the original, potentially hyphenated name of the Gem.

Reply:
@pathawks *I changed it to `specimen` because `foo/bar` is ideally not the given `name`*
But hopefully that should change before this PR gets approved.

Reply:
What I mean is that we should use `name`, with its original value.